src/launcher/java/org/truffleruby/launcher/options/CommandLineOptions.java

```diff
@@ -32,70 +32,57 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 public class CommandLineOptions {
 
-    private Map<String, String> options = new HashMap<>();
+    private final Map<String, CharSequence> options = new HashMap<>();
     private String[] arguments = new String[]{};
     private final List<String> unknownArguments = new ArrayList<>(0);
 
-    // TODO (pitr-ch 26-Jul-2017): move as much as possible to options
-
-    private StringBuffer inlineScript = new StringBuffer();
-    private boolean hasInlineScript;
-    private boolean usePathScript;
-    private String scriptFileName;
-    private boolean showVersion;
-    private boolean showCopyright;
-    private boolean shouldRunInterpreter = true;
-    private boolean shouldPrintUsage;
-    private boolean shouldCheckSyntax;
-    private String inPlaceBackupExtension;
-    private boolean hasScriptToRun; // -e or a file
-    private boolean forceStdin;
-    private boolean shouldPrintShortUsage;
-
     // TODO (pitr-ch 26-Jul-2017): Move to Options when implementing -s option
     // Currently not used
     private Map<String, String> optionGlobals = new HashMap<>();
 
-    public String getDisplayedFileName() {
-        if (isInlineScript()) {
-            if (getScriptFileName() != null) {
-                return getScriptFileName();
-            } else {
-                return "-e";
-            }
-        } else if (shouldUsePathScript()) {
-            return "-S";
-        } else if (isForceStdin() || getScriptFileName() == null) {
-            return "-";
-        } else {
-            return getScriptFileName();
-        }
+    Map<String, CharSequence> getOptionsInternal() {
+        return options;
     }
 
     public Map<String, String> getOptions() {
-        return options;
+        return options.entrySet().stream().
+                collect(Collectors.toMap(Map.Entry::getKey, (e) -> e.getValue().toString()));
```

Comment:
How does `return new HashMap(options)` convert the values from CharSequence to String?